src/features/Dashboard/WeeklySchedule/index.jsx

```diff
@@ -0,0 +1,103 @@
+import React, { useState, useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+
+import { Schedule } from 'react-paragon-topaz';
+
+import { startOfWeek, endOfWeek, isWithinInterval } from 'date-fns';
+
+import { fetchAllClassesData } from 'features/Common/data';
+
+import { formatDateRange } from 'helpers';
+import 'features/Dashboard/WeeklySchedule/index.scss';
+
+const WeeklySchedule = () => {
+  const dispatch = useDispatch();
+  const username = useSelector((state) => state.main.username);
+  const classesData = useSelector((state) => state.common.allClasses.data);
+  const [classList, setClassList] = useState([]);
+  const startCurrentWeek = startOfWeek(new Date());
+  const endCurrentWeek = endOfWeek(new Date());
+  const [stateDate, setStateDate] = useState([
+    {
+      startDate: startCurrentWeek,
+      endDate: endCurrentWeek,
+      key: 'selection',
+      color: '#e4faff',
+    },
+  ]);
+
+  const handleDateChange = (date) => {
+    const startWeekSelected = startOfWeek(date);
+    const endWeekSelected = endOfWeek(date);
+    const newStateSelected = [{
+      startDate: startWeekSelected,
+      endDate: endWeekSelected,
+      key: 'selection',
+    }];
+    setStateDate(newStateSelected);
+  };
+
+  useEffect(() => {
+    if (username) {
+      dispatch(fetchAllClassesData(username));
+    }
+  }, [username, dispatch]);
+
+  useEffect(() => {
+    // Display only the classes which the start date is in the selected time period
+    if (classesData.length > 0) {
+      const classListDisplay = classesData.filter(classItem => {
+        const startDateClass = new Date(classItem.startDate);
+        const startWeekSelected = stateDate[0].startDate;
+        const endWeekSelected = stateDate[0].endDate;
+        return isWithinInterval(startDateClass, {
+          start: startWeekSelected,
+          end: endWeekSelected,
+        });
+      });
+      setClassList(classListDisplay);
+    } else {
+      setClassList([]);
+    }
```

Comment:
This could be more clear in this way:

```suggestion
    if (!classesData.length > 0) {
      setClassList([]);
    }
    
    const classListDisplay = classesData.filter(classItem => {
        const startDateClass = new Date(classItem.startDate);
        const startWeekSelected = stateDate[0].startDate;
        const endWeekSelected = stateDate[0].endDate;
        return isWithinInterval(startDateClass, {
          start: startWeekSelected,
          end: endWeekSelected,
        });
      });
      
      setClassList(classListDisplay);
```